src/HDC1080.c

```diff
@@ -48,7 +50,14 @@ float GetTemperature(void)
 	uint8_t regaddr = HDC1080_TEMP_REG;
 	uint8_t regval[2];
 
-	I2C_IF_ReadFrom(HDC1080_DEV_ADDR, &regaddr, 1, regval, 2);
+	I2C_IF_Write(HDC1080_DEV_ADDR, &regaddr, 1, 0);
+
+
+	// TODO: Change to OS based delay, and add clarity to time param
```

Comment:
I don't know that it's necessary to re-invent the wheel. There are already two different delay functions, one for bare-metal and the other for RTOS based delays. I do admit that we should be using `osi_Sleep( ... )` from the FreeRTOS APIs though. However, I couldn't get that to work, so we will want to readdress this in the future.